Plugin/Sales/Order/View/Details.php

```diff
@@ -0,0 +1,95 @@
+use Magento\Framework\App\RequestInterface;
+use Amwal\Payments\Model\AmwalClientFactory;
+use Amwal\Payments\Model\Config;
+
+class Details
+{
+    protected $authSession;
+    protected $urlBuilder;
+    protected $order;
+    protected $request;
+    protected $amwalClientFactory;
+    private $config;
+
+    public function __construct(
+        Session            $authSession,
+        UrlInterface       $urlBuilder,
+        OrderInterface     $order,
+        RequestInterface   $request,
+        AmwalClientFactory $amwalClientFactory,
+        Config             $config
+    )
+    {
+        $this->authSession = $authSession;
+        $this->urlBuilder = $urlBuilder;
+        $this->order = $order;
+        $this->request = $request;
+        $this->amwalClientFactory = $amwalClientFactory;
+        $this->config = $config;
+    }
+
+    public function beforeSetLayout(View $subject)
+    {
+        $amwalOrderId = $subject->getOrder()->getAmwalOrderId();
+        $amwalClient = $this->amwalClientFactory->create();
+
+        try {
+            $response = $amwalClient->get('transactions/' . $amwalOrderId);
+
+            if ($response->getStatusCode() === 200) {
+                $responseBody = $response->getBody()->getContents();
+                $amwalOrderStatus = json_decode($responseBody)->status;
+
+                $subject->addButton(
+                    'amwal_order_details',
+                    [
+                        'label' => __('Amwal Order Details'),
+                        'class' => $this->isPayValid($subject->getOrder()->getState(), $amwalOrderStatus) ? '' : 'hidden',
+                        'data_attribute' => [
+                            'mage-init' => [
+                                'Amwal_Payments/js/order-details' => [
+                                    'buttonId' => 'amwal_order_details',
+                                    'order_id' => $subject->getOrder()->getId(),
+                                    'amwal_order_id' => $amwalOrderId,
+                                    'order_details' => $responseBody
+                                ]
+                            ]
+                        ],
+                    ]
+                );
+            }
+        } catch (GuzzleException $e) {
+            $message = sprintf(
+                'Unable to set Order details in Amwal for order with ID "%s". Exception: %s',
+                $amwalOrderId,
+                $e->getMessage()
+            );
+            return;
+        }
+    }
+    private function isPayValid($orderState, $amwalOrderStatus)
+    {
+        $defaultOrderStatus = $this->config->getOrderConfirmedStatus();
+
+        if ($orderState === $defaultOrderStatus) {
+            return false;
+        }
+        if($amwalOrderStatus !== 'success') {
+            return false;
+        }
+        $orderStates = [
+            'pending_payment'
+        ];
+        return in_array($orderState, $orderStates);
```

Comment:
No need to check this as an array since it's a single value:
```suggestion
        return $orderState === 'pending_payment';
```